### PR TITLE
Add autoyast support, the SUSE/openSUSE preseed system

### PR DESCRIPTION
**Pull request by ggardet**

### CODE REVIEW

**Comment by BaptisteGerondeau** [comment]
I have succesfully installed mr provisioner (using the Ansible recipes) with Autoyast support using only this additional patch that corrects linting issues : https://github.com/BaptisteGerondeau/mr-provisioner/commit/3dc689c2035255945f7c2380ad51a357461873a7

**Comment by gemagomez** [comment]
@ggardet thanks for your patch! can you please fix the linting issues as per @BaptisteGerondeau 's recommendation?

**Comment by BaptisteGerondeau** [comment]
After some investigation into the travis fail, I've found the reason. Here is the linting command:
prettier --no-semi --write --trailing-comma es5 --single-quote -l 'src/**/*.js' 'src/**/*.jsx'

This will change the files in place.

The version of prettier you use is _paramount_: you need to use the one from package.json, at the moment 1.5.2. If you use the latest one, 1.15.3, although the command goes through (after having used write), npm run lint will always return an error.

https://github.com/BaptisteGerondeau/mr-provisioner/commit/c6a8a6b19f7ab15de09d661b56746d7af724032c

Furthermore, could you please add a tidbit about this patch in the Changelog ?

**Comment by ggardet** [comment]
Done. All check passes.

**Comment by ggardet** [comment]
This PR is ready to merge as tested successfully.

**Comment by gemagomez** [comment]
One last question before merging, for existing deployments, is there anything that needs migrating on the database?

**Comment by ggardet** [comment]
@gemagomez I would say no, but @BaptisteGerondeau probably knows better than me.

**Comment by BaptisteGerondeau** [comment]
@gemagomez To be honest I just ran the ansible installation role and had no issues.
The role does run some upgrading steps [here](https://github.com/Linaro/mr-provisioner-role/blob/ec74e846b85138ab3cb371e5df2ad75db7d901e1/tasks/main.yml#L82). Aside from the role I did not run a single command.

File: mr_provisioner/models.py [hunk 1]
```diff
@@ -934,6 +934,9 @@ def kernel_opts(self, machine, config):
             if config['PRESEED_DNS']:
                 options += " dns=%s" % (config['PRESEED_DNS'])
 
+        elif self.file_type == "autoyast":
+            options = "autoyast=%s" % self.preseed_url(machine, config)
+
```
Comment (gemagomez) [inline mr_provisioner/models.py]: As far as I understand https://www.suse.com/documentation/sles11/singlehtml/book_autoyast/book_autoyast.html#idm139717266766480 it should be possible to set a nameserver based on `config['PRESEED_DNS']` as we do for the others by adding the following to the kernel command line: `nameserver= ...`.

Please add that functionality as well.

Comment (ggardet) [reply mr_provisioner/models.py]: Done

**Comment by rengolin** [comment]
We have tested these changes in production on our lab and they work well. We're closing our side of the work, this patch can now be merged. Thanks!

**Comment by BaptisteGerondeau** [comment]
With the latest changes it installs fine on my MrP VM.
Production test will come later this week.

**Comment by BaptisteGerondeau** [comment]
Tested in a production environment and everything works !
(note: I did not need the nameserver option since it is provided by kea in the lab's case)

**Pull request merged**